spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-nacos-discovery/src/main/java/com/alibaba/cloud/nacos/NacosDiscoveryProperties.java

```diff
@@ -260,17 +275,43 @@ public void init() throws Exception {
 		namespace = Objects.toString(namespace, "");
 		logName = Objects.toString(logName, "");
 
-		if (StringUtils.isEmpty(ip)) {
-			// traversing network interfaces if didn't specify a interface
+		InetUtils.HostInfo hostInfo;
+
+		if (StringUtils.isBlank(host) && StringUtils.isNotBlank(ip)) {
+			host = ip;
+		}
+
+		if (StringUtils.isEmpty(host)) {
 			if (StringUtils.isEmpty(networkInterface)) {
+				hostInfo = inetUtils.findFirstNonLoopbackHostInfo();
+			}
+			else {
+				hostInfo = inetUtils.convertAddress(getInetAddressByNetworkInterface(networkInterface));
+			}
+		}
+		else {
+			try {
+				hostInfo = inetUtils.convertAddress(InetAddress.getByName(host));
+			}
+			catch (UnknownHostException e) {
+				log.error("no such host {}", host, e);
+				throw new IllegalArgumentException("no such host " + host, e);
+			}
+		}
+
+		if (preferIpAddress) {
```

Comment:
I think we don't need field of `preferIpAddress`. It's redundant.

Reply:
This property determines whether we use ip or hostname to register the service。not for config